pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,240 @@
+	// The presence of agentClusterInstall for the local cluster will determine whether or not we need to perform an import
+	agentClusterInstall, err := i.clusterImportOperations.GetAgentClusterInstall(LocalClusterNamespace, LocalClusterNamespace+"-cluster-install")
+	if err != nil && !k8serrors.IsNotFound(err) {
+		i.log.Errorf("hubClusterDay2Importer: There was an error fetching the AgentClusterInstall while checking for need to register local cluster.")
+		return false, err
+	}
+	if agentClusterInstall != nil {
+		i.log.Infof("hubClusterDay2Importer: Found AgentClusterInstall for hub cluster, assuming that hub has been correctly registered for ZTP day 2 operations: %s", agentClusterInstall.Name)
+		return false, nil
+	}
+	return true, nil
+}
+
+func (i *LocalClusterImport) ImportLocalCluster() error {
+	clusterVersion, err := i.clusterImportOperations.GetClusterVersion("version")
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Unable to find cluster version info due to error: %s", err.Error())
+		return err
+	}
+
+	shouldImportLocalCluster, err := i.shouldImportLocalCluster()
+	if err != nil {
+		return err
+	}
+	if !shouldImportLocalCluster {
+		message := "hubClusterDay2Importer: no need to import local cluster as registration already detected"
+		i.log.Info(message)
+		return errors.New(message)
+	}
+	openshift_version := ""
+	release_image := ""
+	if clusterVersion != nil && clusterVersion.Status.History[0].State == configv1.CompletedUpdate {
+		openshift_version = clusterVersion.Status.History[0].Version
+		release_image = clusterVersion.Status.History[0].Image
+		i.log.Infof("hubClusterDay2Importer: Found openshift_version %s", openshift_version)
+	} else {
+		message := "hubClusterDay2Importer: Cluster version info is empty, cannot proceed"
+		i.log.Error(message)
+		return errors.New(message)
+	}
+
+	kubeConfigSecret, err := i.clusterImportOperations.GetSecret("openshift-kube-apiserver", "node-kubeconfigs")
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Unable to fetch local cluster kubeconfigs due to error %s", err.Error())
+		return err
+	}
+	if kubeConfigSecret != nil {
+		i.log.Infof("hubClusterDay2Importer: Found secret %s", kubeConfigSecret.Name)
+	} else {
+		message := "hubClusterDay2Importer: Could not find cluster kubeconfigs for local cluster"
+		i.log.Error(message)
+		return errors.New(message)
+	}
+
+	pullSecret, err := i.clusterImportOperations.GetSecret("openshift-machine-api", "pull-secret")
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Unable to fetch pull secret due to error %s", err.Error())
+		return err
+	}
+	if pullSecret != nil {
+		i.log.Infof("hubClusterDay2Importer: Found secret %s", pullSecret.Name)
+	} else {
+		message := "hubClusterDay2Importer: Could not find pull secret for local cluster"
+		i.log.Error(message)
+		return errors.New(message)
+	}
+
+	dns, err := i.clusterImportOperations.GetDNS("cluster")
```

Comment:
it's a bit strange to have function that gets a const

Reply:
This is easier to mock, the underlying operation can get any DNS entry. I'd rather keep this as is

Reply:
maybe make it GetClusterDNS it will be clearer and you will not need to pass this parameter